google/cloud/aiplatform/tensorboard/tensorboard_resource.py

```diff
@@ -802,6 +815,470 @@ def list(
+                    tensorboard_time_series_id=time_series_id,
+                    value_type=gca_tensorboard_time_series.TensorboardTimeSeries.ValueType.SCALAR,
+                    values=[
+                        gca_tensorboard_data.TimeSeriesDataPoint(
+                            scalar=gca_tensorboard_data.Scalar(value=value),
+                            wall_time=wall_time,
+                            step=step,
+                        )
+                    ],
+                )
+            )
+
+        self.api_client.write_tensorboard_run_data(
+            tensorboard_run=self.resource_name, time_series_data=ts_data
+        )
+
+    def _get_time_series_display_name_to_id_mapping(self) -> Dict[str, str]:
+        """Returns a mapping of the TimeSeries display names to resource IDs for this Run.
+
+        Returns:
+            Dict[str, str] - Dictionary mapping TensorboardTimeSeries display names to
+                resource IDs of TensorboardTimeSeries in this TensorboardRun."""
+        time_series = TensorboardTimeSeries.list(
+            tensorboard_run_name=self.resource_name, credentials=self.credentials
+        )
+
+        return {ts.display_name: ts.name for ts in time_series}
+
+    def _sync_time_series_display_name_to_id_mapping(self):
+        """Updates the local map of TimeSeries diplay name to resource ID."""
+        self._time_series_display_name_to_id_mapping = (
+            self._get_time_series_display_name_to_id_mapping()
+        )
+
+    def create_tensorboard_time_series(
+        self,
+        display_name: str,
+        value_type: Union[
+            gca_tensorboard_time_series.TensorboardTimeSeries.ValueType, str
+        ] = "SCALAR",
+        plugin_name: str = "scalars",
+        plugin_data: Optional[bytes] = None,
+        description: Optional[str] = None,
+    ) -> "TensorboardTimeSeries":
+        """Creates a new tensorboard time series.
+
+        Example Usage:
+
+            tb_ts = tensorboard_run.create_tensorboard_time_series(
+                display_name='my display name',
+                tensorboard_run_name='my-run'
+                tensorboard_id='456'
+                tensorboard_experiment_id='my-experiment'
+                description='my description',
+                labels={
+                    'key1': 'value1',
+                    'key2': 'value2'
+                }
+            )
+
+        Args:
+            display_name (str):
+                Optional. User provided name of this
+                TensorboardTimeSeries. This value should be
+                unique among all TensorboardTimeSeries resources
+                belonging to the same TensorboardRun resource
+                (parent resource).
+            value_type (Union[gca_tensorboard_time_series.TensorboardTimeSeries.ValueType, str]):
+                Optional. Type of TensorboardTimeSeries value. One of 'SCALAR', 'TENSOR', 'BLOB_SEQUENCE'.
+            plugin_name (str):
```

Comment:
Unless this is covered in the docs, might be worth adding more detail to the docstring on what a plugin is.

Reply:
I added additional docstring information from protos. But I don't think this is well documented anywhere. Or at least I couldn't find documentation in the TB docs.